trio_websocket/_impl.py

```diff
@@ -137,14 +168,90 @@ async def open_websocket(
             raise ConnectionTimeout from None
         except OSError as e:
             raise HandshakeError from e
+
+    async def close_connection(connection: WebSocketConnection) -> None:
         try:
-            yield connection
-        finally:
-            try:
-                with trio.fail_after(disconnect_timeout):
-                    await connection.aclose()
-            except trio.TooSlowError:
-                raise DisconnectionTimeout from None
+            with trio.fail_after(disconnect_timeout):
+                await connection.aclose()
+        except trio.TooSlowError:
+            raise DisconnectionTimeout from None
+
+    if _TRIO_MULTI_ERROR:
+        exception_group_type = trio.MultiError  # type: ignore[attr-defined] # pylint: disable=no-member
+    else:
+        exception_group_type = BaseExceptionGroup
```

Comment:
It's repeated.   Consider renaming `_TRIO_MULTI_ERROR` -> `_IS_TRIO_MULTI_ERROR`, and have `_TRIO_MULTI_ERROR` be the multi-error class itself (MultiError or BaseExceptionGroup)